mbtools/copy_resources_s3.py

```diff
@@ -0,0 +1,132 @@
+import hashlib
+import json
+import os
+import io
+import sys
+import magic
+import boto3
+import botocore
+from pathlib import Path
+
+
+def upload_resources(resource_dir, metadata_file, bucket, s3_dir):
+    """Uplaods resources to s3 if they dont already exist there"""
+
+    hash_to_filename_map = new_resource_hashes(resource_dir)
+    hashes_new = list(hash_to_filename_map.keys())
+
+    hashes_metadata = existing_metadata_hashes(metadata_file)
+    hashes_to_update = list(filter(lambda h: h not in hashes_metadata,
+                                   hashes_new)
+                            )
```

Comment:
Minor / subjective: There are various instances of this in the code, but I find this formatting to be a bit hard to read. I like to use a convention where the closing `)` ends up at the same indent level as the original block. So, for example here something like:

```python
    hashes_to_update = list(
        filter(lambda h: h not in hashes_metadata, hashes_new)
    )
    add_new_resources_to_s3(
        bucket, s3_dir, hashes_to_update, hash_to_filename_map, metadata_file
    )
```